packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -245,13 +263,51 @@ export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
               )
             })}
             {pendingInteraction && (
-              <SynapseChatInteraction
-                userMessage={pendingInteraction.userMessage}
-                chatResponseText={pendingInteraction.chatResponseText}
-                chatErrorReason={pendingInteraction.chatErrorReason}
-                progressMessage={currentProgressMessage}
-                scrollIntoView
-              />
+              <>
+                <SynapseChatInteraction
+                  userMessage={pendingInteraction.userMessage}
+                  chatResponseText={pendingInteraction.chatResponseText}
+                  chatErrorReason={pendingInteraction.chatErrorReason}
+                  scrollIntoView
+                />
+                <Box
+                  sx={{
+                    display: 'flex',
+                    flexDirection: 'column',
+                    justifyContent: 'center',
+                  }}
+                >
+                  <SynapseSpinner size={40} />
+                  {/* Show the current message, as well as the full trace log in a tooltip */}
+                  {traceEvents && traceEvents.page && (
+                    <Tooltip
+                      placement="bottom"
+                      title={
+                        <div style={{ textAlign: 'center' }}>
+                          {traceEvents?.page
+                            ?.slice()
+                            .reverse()
+                            .map((event, index) => {
+                              return (
+                                <Typography key={`${index}-${event.message}`}>
+                                  {event.message}
+                                </Typography>
+                              )
+                            })}
+                        </div>
+                      }
+                    >
+                      <Typography
+                        sx={{ textAlign: 'center' }}
+                        variant="body1Italic"
+                      >
+                        {traceEvents.page[0].message}
+                      </Typography>
```

Comment:
Funny that you mentioned that, because I was going down that path already!  It introduced some additional complexity to the PR, so I wanted to implement in a follow-up.

Reply:
The use of a TransitionGroup should really simplify (if it's based on mount/unmount).  This means that I will need to mount/unmount based on the current progress text.  Then it's a matter of positioning (since I don't want the old message to be fading out and taking up vertical space while the new message is fading in)